In [1]:
from read_weather import get_prefectures

prefectures = get_prefectures()
prefectures

{11: '宗谷',
 13: '留萌',
 12: '上川',
 17: '網走・北見・紋別',
 15: '空知',
 16: '後志',
 14: '石狩',
 22: '日高',
 20: '十勝',
 18: '根室',
 24: '檜山',
 21: '胆振',
 19: '釧路',
 23: '渡島',
 31: '青森',
 32: '秋田',
 33: '岩手',
 56: '石川',
 35: '山形',
 34: '宮城',
 55: '富山',
 54: '新潟',
 36: '福島',
 81: '山口',
 68: '島根',
 69: '鳥取',
 63: '兵庫',
 61: '京都',
 57: '福井',
 52: '岐阜',
 48: '長野',
 42: '群馬',
 41: '栃木',
 40: '茨城',
 67: '広島',
 66: '岡山',
 60: '滋賀',
 49: '山梨',
 43: '埼玉',
 84: '長崎',
 85: '佐賀',
 82: '福岡',
 62: '大阪',
 64: '奈良',
 53: '三重',
 51: '愛知',
 50: '静岡',
 46: '神奈川',
 44: '東京',
 45: '千葉',
 86: '熊本',
 83: '大分',
 73: '愛媛',
 72: '香川',
 65: '和歌山',
 88: '鹿児島',
 87: '宮崎',
 74: '高知',
 71: '徳島',
 91: '沖縄',
 99: '南極'}

In [2]:
#import read_weather
#import importlib
#importlib.reload(read_weather)
# pyコード修正時用

from read_weather import get_stations
import time
import pandas as pd

stations = pd.DataFrame()

#for id in prefectures:
for id in [14, 44, 99]:
    pref = prefectures[id]
    print(f"{id}: {pref}")
    pref_st = pd.DataFrame(get_stations(id)).T
    pref_st["prid"] = id
    pref_st["地域名"] = pref
    stations = pd.concat([stations, pref_st])
    time.sleep(5)

stations["緯度"] = stations["北緯"].where(stations["北緯"].notna(), "-" + stations["南緯"])
#stations.to_csv("../files/stations.csv", index=False)
stations

14: 石狩
44: 東京
99: 南極


stid stname  prid kansoku    地点名         カナ        北緯         東経  \
a0029    a0029    阿曽岩    14  100000    阿曽岩       アソイワ  43度18.1分  141度30.8分   
a0030    a0030    新篠津    14  111210    新篠津      シンシノツ  43度13.2分  141度38.7分   
a0031    a0031     山口    14  111200     山口       ヤマグチ  43度08.9分  141度13.3分   
a0032    a0032    手稲山    14  100000    手稲山      テイネヤマ  43度05.0分  141度12.1分   
s47412  s47412     札幌    14  111111     札幌       サツポロ  43度03.6分  141度19.7分   
a0034    a0034    小金湯    14  100010    小金湯       コガネユ  42度57.8分  141度13.0分   
a0035    a0035   恵庭島松    14  111210   恵庭島松    エニワシママツ  42度55.6分  141度33.9分   
a0036    a0036    島松山    14  100000    島松山     シママツヤマ  42度53.7分  141度24.8分   
a0037    a0037   支笏湖畔    14  111200   支笏湖畔     シコツコハン  42度46.3分  141度24.4分   
a1076    a1076    西野幌    14  111100    西野幌     ニシノッポロ  43度02.6分  141度32.4分   
a1085    a1085     石狩    14  111211     石狩       イシカリ  43度11.6分  141度22.2分   
a1193    a1193     浜益    14  111200     浜益       ハママス  43度34.9分  141度23.2分   
a1194    a1194     厚田    14  111210     厚田        アツタ  43度23.8分  141度26.2分   
a1459    a1459     千歳    14  111010     千歳        チトセ  42度46.5分  141度41.5分   
a1507    a1507     江別    14  111200     江別        エベツ  43度06.6分  141度36.1分   
a0365    a0365    小河内    44  111201    小河内       オゴウチ  35度47.5分  139度03.0分   
a0366    a0366    八王子    44  111200    八王子      ハチオウジ  35度40.0分  139度19.0分   
a0367    a0367     調布    44  100000     調布       チョウフ  35度40.1分  139度31.8分   
a0368    a0368    世田谷    44  100000    世田谷       セタガヤ  35度37.6分  139度37.2分   
s47662  s47662     東京    44  111111     東京      トウキヨウ  35度41.5分  139度45.0分   
a0370    a0370  江戸川臨海    44  111200  江戸川臨海   エドガワリンカイ  35度38.3分  139度51.8分   
a0371    a0371     羽田    44  111000     羽田        ハネダ  35度33.2分  139度46.8分   
s47675  s47675     大島    44  111111     大島       オオシマ  34度44.9分  139度21.7分   
s47677  s47677    三宅島    44  111111    三宅島      ミヤケジマ  34度07.4分  139度31.2分   
s47678  s47678    八丈島    44  111111    八丈島      ハチジヨウ  33度07.3分  139度46.7分   
a1001    a1001     青梅    44  111200     青梅        オウメ  35度47.3分  139度18.7分   
a1002    a1002     練馬    44  111200     練馬        ネリマ  35度44.3分  139度35.5分   
a1133    a1133     府中    44  111200     府中       フチュウ  35度41.0分  139度29.0分   
a1134    a1134     新島    44  111100     新島       ニイジマ  34度22.4分  139度15.1分   
a1180    a1180     小沢    44  100000     小沢        オザワ  35度44.1分  139度07.4分   
s47971  s47971     父島    44  111111     父島       チチジマ  27度05.5分  142度11.4分   
a1467    a1467  大島北ノ山    44  111000  大島北ノ山  オオシマキタノヤマ  34度46.9分  139度21.6分   
a1468    a1468  八重見ヶ原    44  111000  八重見ヶ原     ヤエミガハラ  33度06.9分  139度47.1分   
a1505    a1505    神津島    44  100000    神津島      コウヅシマ  34度12.3分  139度08.0分   
a1506    a1506   三宅坪田    44  111000   三宅坪田     ミヤケツボタ  34度04.4分  139度33.6分   
a1509    a1509  三宅島阿古    44  110000  三宅島阿古    ミヤケジマアコ  34度04.8分  139度28.9分   
a1522    a1522     新島    44  111000     新島       ニイジマ  34度22.1分  139度16.1分   
a1523    a1523    神津島    44  111000    神津島      コウヅシマ  34度11.3分  139度08.0分   
a1525    a1525   三宅伊豆    44  110000   三宅伊豆      ミヤケイズ  34度07.2分  139度30.2分   
s47991  s47991    南鳥島    44  111111    南鳥島    ミナミトリシマ  24度17.3分  153度59.0分   
a1595    a1595     母島    44  100000     母島       ハハジマ  26度38.1分  142度09.7分   
a1642    a1642   大島臨時    44  100000   大島臨時    オオシマリンジ  34度44.9分  139度21.7分   
a1643    a1643   大島泉津    44  100000   大島泉津    オオシマセンヅ  34度46.7分  139度25.3分   
a1644    a1644  大島差木地    44  100000  大島差木地   オオシマサシキジ  34度41.0分  139度24.8分   
a1646    a1646     利島    44  100000     利島        トシマ  34度31.7分  139度16.9分   
a1648    a1648    青ヶ島    44  100000    青ヶ島      アオガシマ  32度28.0分  139度45.6分   
s89532  s89532     昭和    99  111111     昭和       ショウワ       NaN   39度34.8分   

            標高  rain   wind   temp    sun   snow    etc 地域名        南緯  \
a0029     320m  True  False  False  False  False  False  石狩       NaN   
a0030       9m  True   True   True   True   True  False  石狩       NaN   
a0031       5m  True   True   True   True  Fal

# 観測所フラグ調査結果
```python
for id, st in stations.items():
    if st["地点名"] == "父島":
        print(id, st["kansoku"])
```
a1646 100000, "降雨"
+ kansoku[0] = 降雨

a1468 111000, "気温", "降雨", "風速"
a1509 110000, "降雨", "風速"
+ kansoku[1] = 風速
+ kansoku[2] = 気温

a0370 111200, "気温", "降雨", "風速", "日照"
s47662 111111, "気温", "降雨", "風速", "日照", "積雪", "その他"
+ kansoku[3] = 日照
+ kansoku[4] = 積雪
+ kansoku[5] = その他

In [3]:
#import read_weather
#import importlib
#importlib.reload(read_weather)
# pyコード修正時用

from read_weather import get_aggrg_periods

get_aggrg_periods()

{9: {'name': '時別値'},
 1: {'name': '日別値'},
 8: {'name': 'N日別値',
  'range': [2,
   3,
   4,
   5,
   6,
   7,
   8,
   9,
   10,
   11,
   12,
   13,
   14,
   15,
   16,
   17,
   18,
   19,
   20,
   21,
   22,
   23,
   24,
   25,
   26,
   27,
   28]},
 2: {'name': '半旬別値'},
 4: {'name': '旬別値'},
 5: {'name': '月別値'},
 6: {'name': '3か月別値'}}

In [4]:
#import read_weather
#import importlib
#importlib.reload(read_weather)
# pyコード修正時用

from read_weather import get_elements

get_elements(8128, 1)

{'201': {'name': '平均気温'},
 '210': {'name': '日平均気温？℃以上日数',
  'options': [40,
   39,
   38,
   37,
   36,
   35,
   34,
   33,
   32,
   31,
   30,
   29,
   28,
   27,
   26,
   25,
   24,
   23,
   22,
   21,
   20,
   19,
   18,
   17,
   16,
   15,
   14,
   13,
   12,
   11,
   10,
   9,
   8,
   7,
   6,
   5,
   4,
   3,
   2,
   1,
   0,
   -1,
   -2,
   -3,
   -4,
   -5,
   -6,
   -7,
   -8,
   -9,
   -10,
   -11,
   -12,
   -13,
   -14,
   -15,
   -16,
   -17,
   -18,
   -19,
   -20,
   -21,
   -22,
   -23,
   -24,
   -25,
   -26,
   -27,
   -28,
   -29,
   -30,
   -31,
   -32,
   -33,
   -34,
   -35,
   -36,
   -37,
   -38,
   -39,
   -40]},
 '204': {'name': '日最高気温の平均'},
 '211': {'name': '日平均気温？℃未満日数',
  'options': [40,
   39,
   38,
   37,
   36,
   35,
   34,
   33,
   32,
   31,
   30,
   29,
   28,
   27,
   26,
   25,
   24,
   23,
   22,
   21,
   20,
   19,
   18,
   17,
   16,
   15,
   14,
   13,
   12,
   11,
   10,
   9,
   8,
   7,
   6,
   5,
   4,
   3,
   2,
   

In [5]:
get_elements(2, 1)

{'201': {'name': '平均気温'},
 '210': {'name': '日平均気温？℃以上日数', 'hidden': 25},
 '204': {'name': '日最高気温の平均'},
 '211': {'name': '日平均気温？℃未満日数', 'hidden': 0},
 '206': {'name': '日最低気温の平均'},
 '212': {'name': '日最高気温？℃以上日数', 'options': [25, 30, 35]},
 '202': {'name': '最高気温'},
 '213': {'name': '日最高気温？℃未満日数', 'hidden': 0},
 '203': {'name': '最低気温'},
 '214': {'name': '日最低気温？℃以上日数', 'hidden': 25},
 '205': {'name': '日最高気温の最低'},
 '215': {'name': '日最低気温？℃未満日数', 'hidden': 0},
 '207': {'name': '日最低気温の最高'},
 '101': {'name': '降水量の合計'},
 '105': {'name': '時間降水量の最大', 'options': [10, 1]},
 '102': {'name': '日降水量の最大'},
 '103': {'name': '日降水量？mm以上日数', 'options': [0, 5, 1, 10, 30, 50, 70, 100]},
 '501': {'name': '最深積雪'},
 '502': {'name': '最深積雪？cm以上の日数', 'options': [0, 5, 10, 20, 50, 100]},
 '503': {'name': '降雪量合計'},
 '504': {'name': '降雪量日合計最大'},
 '401': {'name': '日照時間'},
 '610': {'name': '全天日射量'},
 '402': {'name': '日照率'},
 '403': {'name': '日照時間？時間未満の日数', 'hidden': 0.1},
 '404': {'name': '日照率？%以上の日数', 'hidden': 40},
 '3

In [6]:
get_elements(6, 1)

{'201': {'name': '平均気温'},
 '210': {'name': '日平均気温？℃以上日数', 'hidden': 25},
 '204': {'name': '日最高気温の平均'},
 '211': {'name': '日平均気温？℃未満日数', 'hidden': 0},
 '206': {'name': '日最低気温の平均'},
 '212': {'name': '日最高気温？℃以上日数', 'options': [25, 30, 35]},
 '202': {'name': '最高気温'},
 '213': {'name': '日最高気温？℃未満日数', 'hidden': 0},
 '203': {'name': '最低気温'},
 '214': {'name': '日最低気温？℃以上日数', 'hidden': 25},
 '205': {'name': '日最高気温の最低'},
 '215': {'name': '日最低気温？℃未満日数', 'hidden': 0},
 '207': {'name': '日最低気温の最高'},
 '101': {'name': '降水量の合計'},
 '105': {'name': '時間降水量の最大', 'options': [10, 1, 24]},
 '102': {'name': '日降水量の最大'},
 '103': {'name': '日降水量？mm以上日数', 'options': [0, 5, 1, 10, 30, 50, 70, 100]},
 '501': {'name': '最深積雪'},
 '502': {'name': '最深積雪？cm以上の日数', 'options': [0, 3, 5, 10, 20, 50, 100, 200]},
 '503': {'name': '降雪量合計'},
 '504': {'name': '降雪量日合計最大'},
 '505': {'name': '降雪量日合計？cm以上の日数', 'options': [3, 5, 10, 20, 50, 100]},
 '401': {'name': '日照時間'},
 '610': {'name': '全天日射量'},
 '402': {'name': '日照率'},
 '403': {'name

In [7]:
get_elements(9, 1)

{'201': {'name': '気温'},
 '101': {'name': '降水量'},
 '503': {'name': '降雪の深さ'},
 '501': {'name': '積雪の深さ'},
 '401': {'name': '日照時間'},
 '301': {'name': '風向・風速'},
 '610': {'name': '全天日射量'},
 '601': {'name': '現地気圧'},
 '602': {'name': '海面気圧'},
 '605': {'name': '相対湿度'},
 '604': {'name': '蒸気圧'},
 '612': {'name': '露点温度'},
 '703': {'name': '天気'},
 '607': {'name': '雲量'},
 '704': {'name': '視程'}}

In [8]:
#import read_weather
#import importlib
#importlib.reload(read_weather)
# pyコード修正時用

from read_weather import get_phpsessid, get_weather_as_csv
from datetime import date

element = [201, 202, 203]
station = "s47662"
phpsessid = get_phpsessid()

df = get_weather_as_csv(phpsessid, 1, station, element,
                        date(2022, 7, 1), date(2022, 12, 31))
df

東京                                              
            年月日 平均気温(℃)           最高気温(℃)           最低気温(℃)          
                        品質情報 均質番号         品質情報 均質番号         品質情報 均質番号
0      2022/7/1    30.4    8    1    37.0    8    1    25.4    8    1
1      2022/7/2    29.5    8    1    35.2    8    1    25.4    8    1
2      2022/7/3    28.9    8    1    35.3    8    1    25.2    8    1
3      2022/7/4    26.5    8    1    29.3    8    1    24.4    8    1
4      2022/7/5    26.2    8    1    32.1    8    1    23.9    8    1
..          ...     ...  ...  ...     ...  ...  ...     ...  ...  ...
179  2022/12/27     6.8    8    1    12.3    8    1     2.0    8    1
180  2022/12/28     6.5    8    1    11.9    8    1     2.8    8    1
181  2022/12/29     8.1    8    1    13.7    8    1     2.9    8    1
182  2022/12/30     6.6    8    1    11.8    8    1     2.8    8    1
183  2022/12/31     5.7    8    1    10.5    8    1     3.1    8    1

[184 rows x 10 columns]

In [9]:
if False:
    for year in range(1900, 2023):
        df = download_csv(phpsessid, 1, station, element,
                        date(year, 1, 1), date(year, 12, 31))
        df.to_csv(f"../downloads/tokyo_{year}.csv", index=False)